[sbt 1.0] Make server command the default shell

eed3si9n opened the pull request:
This makes `server` command the default shell.

Note:

There used to be another commit about improving the error handling, but it's moved to another PR - https://github.com/sbt/sbt/pull/3104.

dwijnand reviewed (changes requested):
Let's find some time to talk about this together (perhaps later tonight) because I have the following concerns about making the server default:

* the recent addition about "keeping the shell warm" messaging as well as its implementation
* if perhaps we want to now fold the "server" command into shell, with possibly a "old-shell" fallback command..
* (minor) the fact that sbt-extras invokes sbt explicitly with "shell" while sbt-launcher-package doesn't

I'm happy to sign-off on your Future[Unit] change if you want to tear that out of here into a new PR.

eed3si9n commented:
> I'm happy to sign-off on your Future[Unit] change if you want to tear that out of here into a new PR.

ok. That's split out. https://github.com/sbt/sbt/pull/3104

eed3si9n committed:
server is now called shell

server command is renamed to shell,
and the old shell command is called oldshell.

eed3si9n commented:
After discussion with @dwijnand, the server command is now called `shell`, and the old shell command is called `oldshell` - 86342d0bdd78cc18ee280610742e46a1d434295c

dwijnand commented on main/src/main/scala/sbt/Main.scala:
```diff
@@ -92,7 +92,7 @@ final class xMain extends xsbti.AppMain {
 
   private def notifyUsersAboutShell(state: State) =
-    if (isInteractive && !hasCommand(state, Shell) && !hasCommand(state, Server) && !endsWithBoot(state)) {
+    if (isInteractive && !hasCommand(state, Shell) && !hasCommand(state, OldShell) && !endsWithBoot(state)) {
```
This should be Shell

eed3si9n replied:
It has both Shell and OldShell in the if predicate.

dwijnand replied:
Ah it also has Shell. @eed3si9n if possible could you draw attention to these details when forward-porting?

eed3si9n replied:
There's nothing to forward port since it's sbt 1.0 :)

eed3si9n replied:
Oh you mean when I changed it from the 0.13?

dwijnand replied:
I should've noticed the trailing commits in https://github.com/sbt/sbt/pull/3077
